ui/src/plugin/PluginDetailView.tsx

```diff
@@ -0,0 +1,259 @@
+                        <ConfigurerPanel
+                            pluginInfo={this.state.pluginInfo}
+                            initialConfig={
+                                this.state.currentConfig ? this.state.currentConfig : 'Loading...'
+                            }
+                            save={async (newConfig) => {
+                                await axios.post(
+                                    `${config.get('url')}plugin/${
+                                        this.state.pluginInfo!.id
+                                    }/config`,
+                                    newConfig,
+                                    {headers: {'content-type': 'application/x-yaml'}}
+                                );
+                                await this.refreshFeatures();
+                            }}
+                        />
+                    </PanelWrapper>
+                ) : null}{' '}
+                {this.state.pluginInfo.capabilities.indexOf('displayer') !== -1 ? (
+                    <PanelWrapper
+                        name={'Displayer'}
+                        description={'This is the information generated by the plugin.'}
+                        refresh={this.refreshDisplayer.bind(this)}
+                        icon={Subject}>
+                        <DisplayerPanel
+                            pluginInfo={this.state.pluginInfo}
+                            displayText={this.state.displayText || 'Loading...'}
+                        />
+                    </PanelWrapper>
+                ) : null}
+            </DefaultPage>
+        ) : null;
+    }
+}
+
+class PanelWrapper extends Component<{
+    name: string;
+    description?: string;
+    refresh?: () => Promise<void>;
+    icon?: React.ComponentType;
+}> {
+    public render() {
+        return (
+            <Container style={{display: 'block', width: '100%', margin: '20px', color: 'white'}}>
+                <h1>
+                    {this.props.icon ? (
+                        <span>
+                            <this.props.icon />
+                            &nbsp;
+                        </span>
+                    ) : null}
+                    {this.props.name}
+                    {this.props.refresh ? (
+                        <Button
+                            style={{float: 'right'}}
+                            onClick={() => {
+                                this.props.refresh!();
+                            }}>
+                            <Refresh />
+                        </Button>
+                    ) : null}
+                </h1>
+                {this.props.description ? (
+                    <h4 style={{color: 'grey'}}>{this.props.description}</h4>
+                ) : null}
+                <hr />
+                <div
+                    className={this.props.name
+                        .toLowerCase()
+                        .trim()
+                        .replace(/ /g, '-')}>
```

Comment:
Can be done without regex.
```suggestion
                        .replace(' ', '-')}>
```

Reply:
Then only the first whitespace will be replaced.